linalg/petsc.cpp

```diff
@@ -3819,6 +3816,14 @@ static PetscErrorCode __mfem_ts_ijacobian(TS ts, PetscReal t, Vec x,
    // Jacobian reusage
    ierr = PetscObjectStateGet((PetscObject)P,&ts_ctx->cached_ijacstate);
    CHKERRQ(ierr);
+
+   // Fool DM
+   DM dm;
+   MatType mtype;
+   ierr = MatGetType(P,&mtype); CHKERRQ(ierr);
+   ierr = TSGetDM(ts,&dm); CHKERRQ(ierr);
+   ierr = DMSetMatType(dm,mtype); CHKERRQ(ierr);
+   ierr = DMShellSetMatrix(dm,P); CHKERRQ(ierr);
```

Comment:
Maybe I don't understand the context here, but since this is needed repeatedly it does look like it is a limitation in PETSc right now. Do you want to keep this here as a workaround until there is a better option in PETSc or we do reference counting?

Reply:
The alternative is for MFEM to manage the `DM` , but this requires coding the class `DMMFEM` and it is not a priority now; another aspect is that PETSc uses matrices as containers, which the users can modify by setting values etc, while MFEM concept of getting back the Jacobian is somewhat different (you almost always can get back a different matrix every time). Also, this fix is specific for `DMSHELL` ( which is the default if the user does not provide any) and is a no-op call for any other `DM` type. So it is fine if it stays here (forever....()